doc/cephfs/disaster-recovery-experts.rst

```diff
@@ -316,3 +316,44 @@ operational.
     will contain holes in place of the missing data.
 
 .. _Symbolic link recovery: https://tracker.ceph.com/issues/46166
+
+
+Restoring file system after recovering monitor stores using OSDs
+----------------------------------------------------------------
+
+During rare occasions, all the monitor stores of a cluster may get corrupted.
+To recover the cluster in such a scenario, you need to rebuild the monitor
+stores using the OSDs, and get back the pools intact (active+clean state).
+However, the rebuilt monitor stores don't restore the FS/MDS maps. Additonal
+steps are required to bring back the file system. Currently, only the steps to
+recover a **single active MDS** file system have been identified, and tested.
+These are described below.
+
+Stop all the MDSs of the cluster.
+
+.. warning::
+
+     If the MDSs are not stopped, they may start wiping the pools when
+     recreating the file system.
+
+Recreate the file system using the recovered file system pools.
+
+::
+
+    ceph fs new <fs_name> <metadata_pool> <data_pool> --force
+
+The file system cluster ID, fscid, of the file system will not be preserved.
+To preserve the fscid, pass the desired fscid when recreating the file system.
+
+::
+
+    ceph fs new <fs_name> <metadata_pool> <data_pool> --fscid <fscid> --force
+
+Reset the file system.
+
+::
+
+    ceph fs reset <fs_name> --yes-i-really-mean-it
```

Comment:
Do we know specifically what this is for? Is it necessary? I think resetting a new file system is redundant.

Reply:
It's necessary. I tried restarting the MDSs without executing the steps. While the MDSs do come up, and filesystem status is OK. the data in the file system wasn't retrieved. I think it might be tbecauserank 0 is not set to failed state (which is done by `fs reset` command  https://github.com/ceph/ceph/blob/master/src/mds/FSMap.cc#L503 ) and is in creating state, so when an MDS picks up rank 0 it starts overwriting the metadata instead of fetching the existing metadata?

Reply:
Ah. This seems like an unsatisfactory situation to me. We should provide an option for `fs new` to start rank 0 in a failed state so that this `fs reset` is not necessary. Requiring all MDS to be turned off is onerous in a multifs system.

Let's keep the doc as-is but please add a tracker ticket to fix that.

Reply:
Got it. Let me do this.

Reply:
@batrick created https://tracker.ceph.com/issues/51716
Please check if it looks OK to you.